integration-cli/docker_api_containers_test.go

```diff
@@ -553,3 +553,43 @@ func TestPostContainerBindNormalVolume(t *testing.T) {
 
 	logDone("container REST API - can use path from normal volume as bind-mount to overwrite another volume")
 }
+
+func TestContainerApiPause(t *testing.T) {
+	defer deleteAllContainers()
+	runCmd := exec.Command(dockerBinary, "run", "-d", "busybox", "sleep", "30")
+	out, _, err := runCommandWithOutput(runCmd)
+
+	if err != nil {
+		t.Fatalf("failed to create a container: %s, %v", out, err)
+	}
+	ContainerID := stripTrailingCharacters(out)
+
+	if _, err = sockRequest("POST", "/containers/"+ContainerID+"/pause", nil); err != nil && !strings.Contains(err.Error(), "204 No Content") {
+		t.Fatalf("POST a container pause: sockRequest failed: %v", err)
+	}
+
+	pausedContainers, err := getSliceOfPausedContainers()
+
+	if err != nil {
+		t.Fatalf("error thrown while checking if containers were paused: %v", err)
+	}
+	if len(pausedContainers) != 1 || pausedContainers[0] == "" {
+		t.Fatalf("there should be one paused container and not %d", len(pausedContainers))
+	}
+
+	if _, err = sockRequest("POST", "/containers/"+ContainerID+"/unpause", nil); err != nil && !strings.Contains(err.Error(), "204 No Content") {
+		t.Fatalf("POST a container pause: sockRequest failed: %v", err)
+	}
+
+	pausedContainers, err = getSliceOfPausedContainers()
+
+	if err != nil {
+		t.Fatalf("error thrown while checking if containers were paused: %v", err)
+	}
+
+	if len(pausedContainers) != 0 && len(pausedContainers[0]) != 0 {
```

Comment:
These extra checks you're doing to make sure its not `""` are kind of weird - not your fault :-)  I know.  Would you be interested in modifying `getSliceOfPausedContainers()` so that it never returns an empty string?  Then these if statements would be a little bit easier.